sympy/sets/sets.py

```diff
@@ -165,7 +297,10 @@ def _union(self, other):
 
         Used within the :class:`Union` class
         """
-        return None
+        try:
+            return self._union_simplify(other)
+        except:
```

Comment:
There may be the case when Imageset expr have more than one dummy
            variable present then there may be error in _union_simplify.
            Also it is for simplifying trigonometric eq. solutions (Union of
            Imageset) so one may try with complicated expr in Imageset
            lambda, In that case just return None.

Actually if I remove try block then also it passes all the present testcase.